app/src/main/java/com/bogdan/codeforceswatcher/features/actions/ActionActivity.kt

```diff
@@ -6,43 +6,62 @@ import android.content.Intent
 import android.os.Bundle
 import android.view.KeyEvent
 import android.view.KeyEvent.KEYCODE_BACK
+import android.view.Menu
+import android.view.MenuItem
 import android.view.View
 import android.webkit.WebSettings
 import android.webkit.WebView
 import android.webkit.WebViewClient
 import androidx.appcompat.app.AppCompatActivity
 import com.bogdan.codeforceswatcher.R
-import com.bogdan.codeforceswatcher.features.actions.models.CFAction
 import com.bogdan.codeforceswatcher.store
 import kotlinx.android.synthetic.main.activity_action.*
 
 class ActionActivity : AppCompatActivity() {
 
-    private var cfAction: CFAction? = null
+    private lateinit var pageTitle: String
+    private lateinit var link: String
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_action)
 
-        setSupportActionBar(toolbar)
         supportActionBar?.setDisplayHomeAsUpEnabled(true)
         supportActionBar?.setDisplayShowHomeEnabled(true)
 
+        initData()
+        initViews()
+    }
+
+    private fun initData() {
         val commentId = intent.getLongExtra(COMMENT_ID, -1)
-        cfAction = store.state.actions.actions.find { it.comment?.id == commentId }
+        val cfAction = store.state.actions.actions.find { it.comment?.id == commentId }
+            ?: throw NullPointerException()
 
-        initViews()
+        pageTitle = cfAction.blogEntry.title
```

Comment:
You are initing view here, which isn't obvious as you also have `initViews` method. What I meant in my previous review, that you can pass non-optional `cfAction` to `initViews` method.